### PR TITLE
Added odata filtering support

### PR DESCRIPTION
**Pull request by mingazhev**
Working requests (others will be added later):

`{{ edo.baseUrl  }}/en/api/1.0/availabilities/accommodations/searches/5e700fb0-31b4-4610-b176-9d82c4d777ae?$top=1&$skip=1` - simple top and skip
`{{ edo.baseUrl  }}/en/api/1.0/availabilities/accommodations/searches/5e700fb0-31b4-4610-b176-9d82c4d777ae?$filter=Data/AccommodationDetails/Name eq 'Jade Manotel'` - hotels with name 'Jade Manotel'
`{{ edo.baseUrl  }}/en/api/1.0/availabilities/accommodations/searches/2ee79403-bab5-4359-8a80-027d7fb1a4f6?$filter=Data/RoomContractSets/any(d: d/Price/NetTotal lt 500)` - hotels which contain at least one roomcontract set with price less than 500
...

### CODE REVIEW

File: Api/Models/Accommodations/AvailabilityResult.cs [hunk 1]
```diff
@@ -32,6 +33,6 @@ public AvailabilityResult(AvailabilityResult result, List<RoomContractSet> roomC
         /// <summary>
         /// List of available room contracts sets
         /// </summary>
-        public List<RoomContractSet> RoomContractSets { get; }
+        public IEnumerable<RoomContractSet> RoomContractSets { get; }
```
Comment (kirillta) [inline Api/Models/Accommodations/AvailabilityResult.cs:36]: Is this required?

File: Api/Services/Accommodations/Availability/AvailabilityResultsExtensions.cs [hunk 2]
```diff
@@ -61,9 +43,9 @@ public static class AvailabilityResultsExtensions
         }
 
 
-        private static async Task<List<RoomContractSet>> ProcessRoomContractSetsPrices(List<RoomContractSet> sourceRoomContractSets, PriceProcessFunction priceProcessFunction)
+        private static async Task<List<RoomContractSet>> ProcessRoomContractSetsPrices(IEnumerable<RoomContractSet> sourceRoomContractSets, PriceProcessFunction priceProcessFunction)
         {
-            var roomContractSets = new List<RoomContractSet>(sourceRoomContractSets.Count);
+            var roomContractSets = new List<RoomContractSet>(0);
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/AvailabilityResultsExtensions.cs:48]: Why?

File: Api/Startup.cs [hunk 3]
```diff
@@ -123,11 +124,14 @@ public void ConfigureServices(IServiceCollection services)
...
                 .AddAuthorization()
                 .AddControllersAsServices()
+                .AddMvcOptions(m => m.EnableEndpointRouting = true)
                 .AddFormatterMappings()
                 .AddNewtonsoftJson(options => options.SerializerSettings.Converters.Add(new StringEnumConverter()))
                 .AddApiExplorer()
                 .AddCacheTagHelper()
                 .AddDataAnnotations();
+
+            services.AddOData();
```
Comment (kirillta) [inline Api/Startup.cs:134]: Shouldn't that be placed higher?

File: Api/Models/Accommodations/AvailabilityResult.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Models/Accommodations/AvailabilityResult.cs]: No it isn't, moved back

File: Api/Services/Accommodations/Availability/AvailabilityResultsExtensions.cs (hunk 2 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/AvailabilityResultsExtensions.cs]: To correspond the previous change. Reverted too

File: Api/Startup.cs (hunk 3 above)
Comment (mingazhev) [reply Api/Startup.cs]: I think it doesn't matter. Anyway, moved higher

**Pull request merged**